internal/postgresql/postgresql.go

```diff
@@ -1000,7 +1016,7 @@ func (p *Manager) SyncFromFollowed(followedConnParams ConnParams, replSlot strin
 	return nil
 }
 
-func (p *Manager) RemoveAll() error {
+func (p *Manager) RemoveAllIfInitialized() error {
```

Comment:
Hey @sgotti: so we're now creating two things when you initialise a postgres database, both the data directory and the wal directory.

These two things aren't atomic. Imagine your wal directory is unwriteable, so you create the data directory but not the wal directory.

Bringing this into a method that acknowledges the wal directory may not be deletable, because it was never successfully created, seemed to make sense.

How would you prefer to handle this?

Reply:
Also, I believe this method has always been implemented as a "remove if initialised". This PR hasn't changed the nature of the method, it just changed the name to match what it does.

The only other change is encapsulating the removal of wal+data into a RemoveAll method, to prevent other places in the codebase from forgetting to handle the wal directory. Would you prefer to drop this method and copy and paste its contents instead?

Reply:
Actually this is slightly wrong- this should probably be `RemoveAllUnlessStarted`. We can change the name to match, pending your feedback.

Reply:
This method was meant to just remove all the postgres data managed by stolon. I can't understand why it should check if the instance is initialized (this check is done by the keeper). RemoveAll should just remove all data and also the wal dir if defined. Failing to remove a wal dir should be IMHO considered an error since it is just another directory managed by stolon.